t/load_rexfile.t

```diff
@@ -0,0 +1,224 @@
+## no critic (Modules);
+use 5.010001;
+use strict;
+use warnings;
+
+use Test::More;
+use Test::Output;
+use File::Temp;
+use File::Spec;
+
+use Rex::CLI;
+## no critic (RequireTrailingCommaAtNewline);
+## no critic (ProhibitPostfixControls, WhileDiamondDefaultAssignment);
+## no critic (ProhibitPunctuationVars, ProhibitPackageVars);
+## no critic (RequireCheckedSyscalls, RequireCheckedClose);
+## no critic (RegularExpressions);
+## no critic (Carping, ProhibitNoWarnings, DuplicateLiteral);
+
+#diag 'create some rexfiles to test...';
+my $fh      = undef;
+my $testdir = File::Temp->newdir( 'rextest.XXXX', TMPDIR => 1, CLEANUP => 1 );
+while (<DATA>) {
+  last if /^__END__$/;
+  if (/^@@ *(\S+)$/) {
+
+    #diag "prepare file $1";
+    close $fh if $fh;
+    open $fh, '>', File::Spec->catfile( $testdir, $1 ) or die $!;
+    next;
+  }
+  print {$fh} $_ if $fh;
+}
+close $fh if $fh;
+
+our $exit_was_called = undef;
+
+# we must disable Rex::CLI::exit() sub imported from Rex::Commands
+no warnings 'redefine';
+local *Rex::CLI::exit = sub { $exit_was_called = 1 };
+use warnings 'redefine';
```

Comment:
I don't seem to get any warnings even without `no warnings 'redefine'` so it might be not needed after all :thinking: 

This part might need further review to see if we can avoid adding further `no warnings` to the code base, and/or if there's another way we can take than redefining subroutines.

Reply:
Yeah, I'm familiar with that message, that's why it baffles me even more that it does not show up on my system :thinking: I'll cross-check the behavior later in my private CI and GitHub Actions environments as well.

One way I'm thinking to check is testing for the exit/exit code instead of the redefine (perhaps something like `Test::Exit` could help). Another one is to see if we can restructure the `load_rexfile()` to make it easier to test; for example let it return failure in other ways than `exit` (throw an exception we may catch, or return something useful, etc.).

Reply:
As [commented below](https://github.com/RexOps/Rex/pull/1522#issuecomment-1399320498), I did not see the redefinition warnings on my systems because I used perl-5.36.0, where

> Localized subroutine redefinitions no longer trigger this warning.

(source: [Changes to Existing Diagnostics](https://metacpan.org/release/RJBS/perl-5.36.0/view/pod/perldelta.pod#Changes-to-Existing-Diagnostics) section of perldelta for perl-5.36.0)

I've decided to use Sub::Override to override functions in a clean way (and with some extra protection against typos, etc.). I expect that would be useful elsewhere in tests, and should not be a hassle for downstream packagers to include.